src/rules/no-unnecessary-reactive-curlies.ts

```diff
@@ -0,0 +1,50 @@
+import type { AST } from "svelte-eslint-parser"
+import { createRule } from "../utils"
+
+export default createRule("no-unnecessary-reactive-curlies", {
+  meta: {
+    docs: {
+      description:
+        "disallow wrapping single reactive statements in curly braces",
+      category: "Stylistic Issues",
+      recommended: false,
+      conflictWithPrettier: true,
+    },
+    fixable: "code",
+    schema: [],
+    messages: {
+      unnecessaryCurlies: `Do not wrap reactive statements in curly braces unless necessary.`,
+    },
+    type: "suggestion",
+  },
+  create(context) {
+    return {
+      [`SvelteReactiveStatement > BlockStatement[body.length=1]`]: (
+        node: AST.SvelteReactiveStatement,
+      ) => {
+        const source = context.getSourceCode()
+
+        return context.report({
+          node,
+          loc: node.loc,
+          messageId: "unnecessaryCurlies",
+
+          fix(fixer) {
```

Comment:
I think it is necessary to stop the autofix in the edge case that may break the source code.

```svelte
<script>
    $: { let name = 'world'; }
    $: { function fn(){} }
    $: { class MyClass {} }
</script>
```

Reply:
Added a check for the body statement to ensure that it's an assignment expression.